Clean and refactor scripts in toolkit.html

ericvennemeyer opened the pull request:
Fixes #2172

### What changes did you make and why did you make them ?

  - Modified top script in `toolkit.html`
      - Added event listener on `DOMContentLoaded` to filter cards by category as soon as page loads
      - This fixes the bug mentioned in the original issue
      - Added spaces between functions to improve readability
  - Created new file `toolkit.js` in `assets/js` and moved top script from `toolkit.html` into it
  - Deleted bottom script in `toolkit.html` that added `margin-bottom: 3.5rem` to divs with class `toolkit-info-container` that contained "status text"
  - In `_sass/components/_toolkit.scss`:
      - Changed style for `toolkit-info-container` from `margin: 1rem 0rem 2rem` to `margin: 1rem 0rem 3.5rem`
      - This replicates the spacing applied by the old script

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)
<!-- Note, if your images are too big, use the <img src="" width="" length="" />  syntax instead of ![image](link) to format the images -->
<!-- If images are not loading properly, you might need to double check the syntax or add a newline after the closing </summary> tag -->

<details>
<summary>Visuals before changes are applied</summary>

<img width="2078" alt="Before" src="https://user-images.githubusercontent.com/82679637/186496995-0eadd32a-070a-4801-b9a9-841498e7aec8.png">

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
<img width="2067" alt="After" src="https://user-images.githubusercontent.com/82679637/186497048-606a870e-3398-4472-9b22-5bb933a73bb7.png">

</details>

ericvennemeyer committed:
Clean up scripts in toolkit.html (#2172)

ericvennemeyer committed:
Clean up scripts in toolkit.html, delete commented-out script (#2172)

ericvennemeyer commented:
Hey @kathrynsilvaconway @blulady @Wny-Duong Just opened this PR and ran into linter trouble. It's complaining because there are some repeated selectors in the `_sass/components/_toolkit.scss` file. Weirdly enough, it looks like they were already there in `gh-pages`, and I don't _think_ I caused it.

I **did** mess up by forgetting to delete some commented-out code, and ended up submitting two commits in my PR . . . but I don't think that is related to this, is it?

blulady commented:
> Hey @kathrynsilvaconway @blulady @Wny-Duong Just opened this PR and ran into linter trouble. It's complaining because there are some repeated selectors in the `_sass/components/_toolkit.scss` file. Weirdly enough, it looks like they were already there in `gh-pages`, and I don't _think_ I caused it.
> 
> I **did** mess up by forgetting to delete some commented-out code, and ended up submitting two commits in my PR . . . but I don't think that is related to this, is it?

Good Afternoon Eric,
if you click on details next to the failed linter warning it will direct you to what is causing this error. 
![Screenshot (224)](https://user-images.githubusercontent.com/67162265/186504109-b8b4ca09-b563-41ca-a402-c062712f2bbf.jpg)

ericvennemeyer commented:
Hi @blulady thanks for the quick reply. I did already check out the details in the linter warning. See my previous comment:

> It's complaining because there are some repeated selectors in the _sass/components/_toolkit.scss file.

As far as I can tell, it's being caused by a mistake that's already in the code in `gh-pages`, and not by something that I did. I'm confused about that and not certain how to move forward.

blulady commented:
> Hi @blulady thanks for the quick reply. I did already check out the details in the linter warning. See my previous comment:
> 
> > It's complaining because there are some repeated selectors in the _sass/components/_toolkit.scss file.
> 
> As far as I can tell, it's being caused by a mistake that's already in the code in `gh-pages`, and not by something that I did. I'm confused about that and not certain how to move forward.

if it's our repo wouldn't it be happening for all pr's... can you create a few other pr's to check?

ericvennemeyer commented:
That's what I'm confused about. I'll open a few test PRs--maybe one with only a comment added to` _toolkit.scss`, and another with no changes at all as a control--and let you know what happens.

blulady commented:
> That's what I'm confused about. I'll open a few test PRs--maybe one with only a comment added to` _toolkit.scss`, and another with no changes at all as a control--and let you know what happens.

My PR passed the linter, I think something's going on in your local Repo...

ericvennemeyer commented:
Did your test PR have changes in the `_toolkit.scss` file specifically?

I just ran my tests and was able to repeat the error. I created two new topic branches as follows:

- `linter-test-toolkit-scss`: added a comment to the top of `_toolkit.scss`, otherwise no change from `gh-pages`
- `linter-test-toolkit-html`: added a comment to the top of `toolkit.html`, otherwise no change from `gh-pages`

When attempting to open PRs for each, the first one threw the same linter error as my original PR, while the second one passed.

blulady commented:
> Did your test PR have changes in the `_toolkit.scss` file specifically?
> 
> I just ran my tests and was able to repeat the error. I created two new topic branches as follows:
> 
> * `linter-test-toolkit-scss`: added a comment to the top of `_toolkit.scss`, otherwise no change from `gh-pages`
> * `linter-test-toolkit-html`: added a comment to the top of `toolkit.html`, otherwise no change from `gh-pages`
> 
> When attempting to open PRs for each, the first one threw the same linter error as my original PR, while the second one passed.

I updated my local before creating the branch and the pr. Try updating your gh-pages and then merge it into the branch you are testing.

ericvennemeyer commented:
I also did that before creating my test branches. `git fetch upstream` doesn't return any new changes, and the topic branches were all created from an up-to-date `gh-pages` to begin with. 

So you're saying you opened a PR that included changes to the `_toolkit.scss` file, and it passed the linter? If that's the case, I may just need to delete and refork my Website repo and start everything over from there. (Although, the line numbers the linter is complaining about in my PR are also present in the `hackforla/website` main repo. So it's definitely true that there are duplicate CSS selectors in `_toolkit.scss`.)

ericvennemeyer commented:
FYI I looked back at the last time someone pushed a commit that included `_toolkit.scss` and @Wny-Duong experienced the same linter error I did:
https://github.com/hackforla/website/pull/3432#pullrequestreview-1054953613

blulady commented:
> I also did that before creating my test branches. `git fetch upstream` doesn't return any new changes, and the topic branches were all created from an up-to-date `gh-pages` to begin with.
> 
> So you're saying you opened a PR that included changes to the `_toolkit.scss` file, and it passed the linter? If that's the case, I may just need to delete and refork my Website repo and start everything over from there. (Although, the line numbers the linter is complaining about in my PR are also present in the `hackforla/website` main repo. So it's definitely true that there are duplicate CSS selectors in `_toolkit.scss`.)
No, I am I did not do those things. I know the linter fail is related to things you are doing in your repo though. It probably shouldn't be flagging comments. Feel free to make it an issue and then research how to change the linter so it's not flagging comments.

ericvennemeyer commented:
> No, I am I did not do those things.

Sorry, I'm not trying to be difficult. But I'm not sure I understood your last comment. What do you mean by this?

> It probably shouldn't be flagging comments. Feel free to make it an issue and then research how to change the linter so it's not flagging comments.

It's not flagging the comments. it's flagging three instances of CSS selectors that are duplicated. Here's a screenshot of the error from my PR, followed by one from the test PR where I only added a comment to `_toolkit.scss` and made no other changes:

<img width="916" alt="original PR" src="https://user-images.githubusercontent.com/82679637/186523167-c47cb87d-92e6-485f-b1b0-2df981a35a47.png">

<img width="933" alt="test PR" src="https://user-images.githubusercontent.com/82679637/186523189-7162f6cc-6b46-4395-800d-429a20abc665.png">

(The difference in line numbers in the second screenshot is caused by the single-line comment, plus a blank newline, that I added to the test branch.)

blulady commented:
> > No, I am I did not do those things.
> 
> Sorry, I'm not trying to be difficult. But I'm not sure I understood your last comment. What do you mean by this?
> 
> > It probably shouldn't be flagging comments. Feel free to make it an issue and then research how to change the linter so it's not flagging comments.
> 
> It's not flagging the comments. it's flagging three instances of CSS selectors that are duplicated. Here's a screenshot of the error from my PR, followed by one from the test PR where I only added a comment to `_toolkit.scss` and made no other changes:
> 
> <img alt="original PR" width="916" src="https://user-images.githubusercontent.com/82679637/186523167-c47cb87d-92e6-485f-b1b0-2df981a35a47.png">
> 
> <img alt="test PR" width="933" src="https://user-images.githubusercontent.com/82679637/186523189-7162f6cc-6b46-4395-800d-429a20abc665.png">
> 
> (The difference in line numbers in the second screenshot is caused by the single-line comment, plus a blank newline, that I added to the test branch.)

Ok then make an issue about that instead.

ericvennemeyer commented:
Will do

kathrynsilvaconway commented:
ETA: End of day 8/27
Availability: 2 hours

giroz commented:
ETA: 8-28-22
Availability:  3 hours

ericvennemeyer commented:
FYI the duplicate CSS selectors in `_toolkit.scss` that caused the `Lint SCSS` test to fail are now addressed in issue #3489.

kathrynsilvaconway reviewed (approved):
@ericvennemeyer Really nice work on this issue. I can see the changes you made locally and all the URLs work. There are no unexpected results from your changes.

As per our discussion, I am satisfied that the linting failure is happening because there are duplicate selectors in the existing code base. I ran a test myself and got the same errors. We are addressing this in a separate issue, so there is nothing else to be done about it here. I am approving your changes.

ericvennemeyer commented:
@kathrynsilvaconway Thank you for reviewing!

arpitapandya commented:
ETA: EOD 2022-08-28
Availability: 2 hours

arpitapandya reviewed (approved):
All the changes looks correct! Great job @ericvennemeyer , Though, I am not running into any linter errors.

giroz commented:
I'm gonna say great job, Eric.  I having trouble with my git and going to have ask someone.  Since you got Arpita and Kathryn signed out this you got it.

ericvennemeyer commented:
Thanks @arpitapandya and @giroz!!
